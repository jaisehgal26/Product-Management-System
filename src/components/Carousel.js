import React from "react";
import carousel_1 from "../img/carousel-1.jpg";
import carousel_2 from "../img/carousel-2.jpg";
import Fade from "react-reveal/Fade";
import { Link } from "react-router-dom";

import "../App.css";

export default function Carousel() {
  return (
    <>
      <div
        id="carouselExampleCaptions"
        className="carousel slide"
        data-bs-ride="carousel"
      >
        <div className="carousel-indicators">
          <button
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to="0"
            className="active"
            aria-current="true"
            aria-label="Slide 1"
          ></button>
          <button
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to="1"
            aria-label="Slide 2"
          ></button>
        </div>
        <div className="carousel-inner">
          <div className="carousel-item active">
            <img
              src={carousel_1}
              className="d-block w-100"
              alt="..."
              style={{ filter: " blur(1px)" }}
            />
            <div className="carousel-caption d-none d-md-block">
              <div className="container">
                <div className="row justify-content-start">
                  <Fade top>
                    <div className="col my-5">
                      <p className="carousel_header text-uppercase fw-bold mb-2">
                        #The Best Bakery
                      </p>
                      <h1 className="display-1  mb-4 animated slideInDown carousel_detail">
                        We Bake With Passion
                      </h1>
                      <blockquote className="blockquote mb-4">
                        <p>
                          <i className="fas fa-quote-left fa-lg text-warning me-2"></i>
                          <span className="font-italic">
                            "Baking cookies is comforting, they are the sweetest
                            little bit of comfort food. They are very bite-sized
                            and personal.""
                          </span>
                        </p>
                      </blockquote>

                      <Link
                        to="/about"
                        className="carousel_btn btn rounded-pill py-3 px-5"
                      >
                        Read More
                      </Link>
                    </div>
                  </Fade>
                </div>
              </div>
            </div>
          </div>
          <div className="carousel-item">
            <img
              src={carousel_2}
              className=" d-block w-100"
              alt="..."
              style={{ filter: " blur(1px)" }}
            />
            <div className="carousel-caption d-none d-md-block">
              <div className="container">
                <div className="row justify-content-start">
                  <Fade top>
                    <div className="col my-5">
                      <p className="carousel_header text-uppercase fw-bold mb-2">
                        #The Best Bakery
                      </p>
                      <h1 className="display-1  mb-4 animated slideInDown carousel_detail">
                        We Bake With Passion
                      </h1>
                      <blockquote className="blockquote mb-4">
                        <p>
                          <i className="fas fa-quote-left fa-lg text-warning me-2"></i>
                          <span className="font-italic">
                            "Baking cookies is comforting, they are the sweetest
                            little bit of comfort food. They are very bite-sized
                            and personal."
                          </span>
                        </p>
                      </blockquote>

                      <Link
                        to="/about"
                        className="carousel_btn btn rounded-pill py-3 px-5"
                      >
                        Read More
                      </Link>
                    </div>
                  </Fade>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          className="carousel-control-prev"
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide="prev"
        >
          <span
            className="carousel-control-prev-icon"
            aria-hidden="true"
          ></span>
          <span className="visually-hidden">Previous</span>
        </button>
        <button
          className="carousel-control-next"
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide="next"
        >
          <span
            className="carousel-control-next-icon"
            aria-hidden="true"
          ></span>
          <span className="visually-hidden">Next</span>
        </button>
      </div>
    </>
  );
}
