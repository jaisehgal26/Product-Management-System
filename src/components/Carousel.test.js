import * as React from "react";

import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Carousel from "./Carousel";

test("should render correctly", () => {
  const wrapper = toJson(shallow(<Carousel />));
  expect(wrapper).toMatchSnapshot();
});
it("includes two paragraphs", () => {
  const wrapper = shallow(<Carousel />);
  expect(wrapper.find("p").length).toEqual(4);
});
