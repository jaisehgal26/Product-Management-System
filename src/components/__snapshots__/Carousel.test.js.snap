// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Fragment>
  <div
    className="carousel slide"
    data-bs-ride="carousel"
    id="carouselExampleCaptions"
  >
    <div
      className="carousel-indicators"
    >
      <button
        aria-current="true"
        aria-label="Slide 1"
        className="active"
        data-bs-slide-to="0"
        data-bs-target="#carouselExampleCaptions"
        type="button"
      />
      <button
        aria-label="Slide 2"
        data-bs-slide-to="1"
        data-bs-target="#carouselExampleCaptions"
        type="button"
      />
    </div>
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item active"
      >
        <img
          alt="..."
          className="d-block w-100"
          src="carousel-1.jpg"
          style={
            Object {
              "filter": " blur(1px)",
            }
          }
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <div
            className="container"
          >
            <div
              className="row justify-content-start"
            >
              <Fade
                top={true}
              >
                <div
                  className="col my-5"
                >
                  <p
                    className="carousel_header text-uppercase fw-bold mb-2"
                  >
                    #The Best Bakery
                  </p>
                  <h1
                    className="display-1  mb-4 animated slideInDown carousel_detail"
                  >
                    We Bake With Passion
                  </h1>
                  <blockquote
                    className="blockquote mb-4"
                  >
                    <p>
                      <i
                        className="fas fa-quote-left fa-lg text-warning me-2"
                      />
                      <span
                        className="font-italic"
                      >
                        "Baking cookies is comforting, they are the sweetest little bit of comfort food. They are very bite-sized and personal.""
                      </span>
                    </p>
                  </blockquote>
                  <Link
                    className="carousel_btn btn rounded-pill py-3 px-5"
                    to="/about"
                  >
                    Read More
                  </Link>
                </div>
              </Fade>
            </div>
          </div>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="..."
          className=" d-block w-100"
          src="carousel-2.jpg"
          style={
            Object {
              "filter": " blur(1px)",
            }
          }
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <div
            className="container"
          >
            <div
              className="row justify-content-start"
            >
              <Fade
                top={true}
              >
                <div
                  className="col my-5"
                >
                  <p
                    className="carousel_header text-uppercase fw-bold mb-2"
                  >
                    #The Best Bakery
                  </p>
                  <h1
                    className="display-1  mb-4 animated slideInDown carousel_detail"
                  >
                    We Bake With Passion
                  </h1>
                  <blockquote
                    className="blockquote mb-4"
                  >
                    <p>
                      <i
                        className="fas fa-quote-left fa-lg text-warning me-2"
                      />
                      <span
                        className="font-italic"
                      >
                        "Baking cookies is comforting, they are the sweetest little bit of comfort food. They are very bite-sized and personal."
                      </span>
                    </p>
                  </blockquote>
                  <Link
                    className="carousel_btn btn rounded-pill py-3 px-5"
                    to="/about"
                  >
                    Read More
                  </Link>
                </div>
              </Fade>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="carousel-control-prev"
      data-bs-slide="prev"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="visually-hidden"
      >
        Previous
      </span>
    </button>
    <button
      className="carousel-control-next"
      data-bs-slide="next"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="visually-hidden"
      >
        Next
      </span>
    </button>
  </div>
</Fragment>
`;
